docs/source/features/memory_optimizations.rst

```diff
@@ -5,6 +5,80 @@ Parallelism
 -----------
 Refer to :doc:`Parallelism <./parallelisms>`.
 
+
+Mixture of Experts
+------------------
+
+Overview
+^^^^^^^^
+
+NeMo supports Mixture of Experts (MoE)in the transformer layer for NLP models.
+
+MoE is a machine learning technique where multiple specialized models (experts,
+usually multi-layer perceptrons) are combined to solve a complex task. Each expert
+focuses on a specific subtask or domain, while a gating network dynamically activates
+the most appropriate expert based on the current input.
+
+
+To use MoE  in the NeMo Framework, adjust the ``num_moe_experts`` parameter in the model configuration:
+
+1. Set ``num_moe_experts`` to `8` to leverage 8 experts in the MoE module.
+
+   .. code-block:: yaml
+
+       num_moe_experts: 8  # Set MoE to use 8 experts
+
+2. Set ``moe_router_topk`` to the number of experts you want activated, for eaxmple if you want to process each input with two experts:
```

Comment:
Fix spelling abd add sentence.

2. Set ``moe_router_topk`` to the number of experts you want activated. For example, if you want to process each input with two experts: